### ISSUE TITLE
ci.yaml changes aren't applied before tests run post-submit

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#95037]
A PR like https://github.com/flutter/flutter/pull/93086 that makes significant changes to `ci.yaml` will pass presubmits but fail postsubmits because of some delay in the infrastructure.

The scheduler should avoid scheduling tests when there's a pending infrastructure change that needs to roll out for the tests to pass.

**Comment by CaseyHillers** [comment]
This is blocked on https://github.com/flutter/flutter/issues/81795

Once that's implemented, we'll roll out the cocoon scheduler to be used by all the Flutter repos instead of the LUCI scheduler. ETA is migration will be complete by end of January.

**Comment by godofredoc** [comment]
Moving as P3, this is a nice to have but it is not currently blocking anything.

**Comment by CaseyHillers** [comment]
Cocoon scheduler migration is complete, and everything should be tested based on what's in the ci.yaml for that repo.
